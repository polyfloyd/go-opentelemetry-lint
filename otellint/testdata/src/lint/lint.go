package lint

import (
	"context"
	"database/sql"

	"go.opentelemetry.io/otel"
	"go.opentelemetry.io/otel/trace"
)

const tracerName = "testdata/lint"

func tracer() trace.Tracer {
	return otel.Tracer("testdata/lint")
}

func AddToContext(ctx context.Context, thing string) context.Context {
	return context.WithValue(ctx, 1337, thing)
}

func AddToContextButCanFail(ctx context.Context, thing string) (context.Context, error) {
	return context.WithValue(ctx, 1337, thing), nil
}

func GetFromContext(ctx context.Context) string {
	return ctx.Value(1337).(string)
}

func SpanOk(ctx context.Context, db *sql.DB) error {
	ctx, span := tracer().Start(ctx, "SpanOk")
	defer span.End()

	row := db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

func SpanOkOtherStyle(ctx context.Context, db *sql.DB) error {
	ctx, span := otel.Tracer(tracerName).Start(ctx, "SpanOkOtherStyle")
	defer span.End()

	row := db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

func SpanMisspelled(ctx context.Context, db *sql.DB) error {
	ctx, span := tracer().Start(ctx, "queryThing") // want "OpenTelemetry span misspelled, expected `SpanMisspelled`"
	defer span.End()

	row := db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

type querier struct {
	db *sql.DB
}

func (q *querier) Query(ctx context.Context) error { // want "Missing OpenTelemetry span"
	row := q.db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

func (q *querier) Query2(ctx context.Context) error {
	ctx, span := tracer().Start(ctx, "queryThing") // want "OpenTelemetry span misspelled, expected `\\(\\*querier\\).Query2`"
	defer span.End()

	row := q.db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

func (q *querier) Query3(c context.Context) error { // want "Missing OpenTelemetry span"
	row := q.db.QueryRowContext(c, `SELECT * FROM sample_text`)
	return row.Err()
}
