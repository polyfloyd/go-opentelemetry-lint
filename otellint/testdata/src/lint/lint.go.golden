package lint

import (
	"context"
	"database/sql"
	"fmt"
	"net/http"

	"go.opentelemetry.io/otel"
	"go.opentelemetry.io/otel/trace"
)

const tracerName = "testdata/lint"

func tracer() trace.Tracer {
	return otel.Tracer("testdata/lint")
}

func AddToContext(ctx context.Context, thing string) context.Context {
	return context.WithValue(ctx, 1337, thing)
}

func AddToContextButCanFail(ctx context.Context, thing string) (context.Context, error) {
	return context.WithValue(ctx, 1337, thing), nil
}

func GetFromContext(ctx context.Context) string {
	return ctx.Value(1337).(string)
}

func SpanOk(ctx context.Context, db *sql.DB) error {
	ctx, span := tracer().Start(ctx, "SpanOk")
	defer span.End()

	row := db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

func SpanOkOtherStyle(ctx context.Context, db *sql.DB) error {
	ctx, span := otel.Tracer(tracerName).Start(ctx, "SpanOkOtherStyle")
	defer span.End()

	row := db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

func SpanMisspelled(ctx context.Context, db *sql.DB) error {
	ctx, span := tracer().Start(ctx, "SpanMisspelled") // want "OpenTelemetry span misspelled, expected `SpanMisspelled`"
	defer span.End()

	row := db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

type querier struct {
	db *sql.DB
}

func (q *querier) Query(ctx context.Context) error { // want "Missing OpenTelemetry span"
	ctx, span := tracer().Start(ctx, "(*querier).Query")
	defer span.End()

	row := q.db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

func (q *querier) Query2(ctx context.Context) error {
	ctx, span := tracer().Start(ctx, "(*querier).Query2") // want "OpenTelemetry span misspelled, expected `\\(\\*querier\\).Query2`"
	defer span.End()

	row := q.db.QueryRowContext(ctx, `SELECT * FROM sample_text`)
	return row.Err()
}

func (q *querier) Query3(c context.Context) error { // want "Missing OpenTelemetry span"
	c, span := tracer().Start(c, "(*querier).Query3")
	defer span.End()

	row := q.db.QueryRowContext(c, `SELECT * FROM sample_text`)
	return row.Err()
}

func HTTPHanderMismatched(w http.ResponseWriter, r *http.Request) {
	ctx, span := tracer().Start(r.Context(), "HTTPHanderMismatched") // want "OpenTelemetry span misspelled"
	defer span.End()
	_ = ctx

	fmt.Fprintf(w, "hi")
}
func HTTPHanderMissingSpan(w http.ResponseWriter, r *http.Request) { // want "Missing OpenTelemetry span"
	ctx, span := tracer().Start(r.Context(), "HTTPHanderMissingSpan")
	defer span.End()

	fmt.Fprintf(w, "hi")
}
